test/expr/unary/keypath/keypath.swift

```diff
@@ -229,7 +230,9 @@ func testDisembodiedStringInterpolation(x: Int) {
 
 func testNoComponents() {
   let _: KeyPath<A, A> = \A // expected-error{{must have at least one component}}
+  // FIXME(diagnostics): This should be diagnosed as `missing generic parameter 'T'` instead of a contextual failure.
   let _: KeyPath<C, A> = \C // expected-error{{must have at least one component}} expected-error{{}}
```

Comment:
@hborla This is something that https://github.com/apple/swift/pull/27443 would helpfully address 🤞